lib/experimental/Navigation/Sidebar/Sidebar.tsx

```diff
@@ -14,10 +37,35 @@ export function Sidebar({ header, body, footer }: SidebarProps) {
   const { sidebarState, isSmallScreen } = useSidebar()
   const shouldReduceMotion = useReducedMotion()
   const [isScrolled, setIsScrolled] = useState(false)
+  const [hasScrollBottom, setHasScrollBottom] = useState(false)
+  const topScrollDetectorRef = useRef<HTMLDivElement>(null)
+  const bottomScrollDetectorRef = useRef<HTMLDivElement>(null)
 
-  const handleScroll = (e: React.UIEvent<HTMLDivElement>) => {
-    setIsScrolled(e.currentTarget.scrollTop > 0)
-  }
+  useEffect(() => {
+    const observerOptions = { threshold: [1] }
+
+    const createObserver = (callback: (isIntersecting: boolean) => void) => {
+      return new IntersectionObserver(
+        ([entry]) => callback(!entry.isIntersecting),
+        observerOptions
+      )
+    }
+
+    const topObserver = createObserver(setIsScrolled)
+    const bottomObserver = createObserver(setHasScrollBottom)
+
+    if (topScrollDetectorRef.current) {
+      topObserver.observe(topScrollDetectorRef.current)
+    }
+    if (bottomScrollDetectorRef.current) {
+      bottomObserver.observe(bottomScrollDetectorRef.current)
+    }
+
+    return () => {
+      topObserver.disconnect()
+      bottomObserver.disconnect()
+    }
+  }, [])
```

Comment:
Great work! But - you have this fantastic tool at your disposal `useIntersectionObserver`. It's in the `usehooks-ts` dependency 😉 

https://usehooks-ts.com/react-hook/use-intersection-observer